From 9218b57cdfb08046ae6278180364e13b7082297c Mon Sep 17 00:00:00 2001
From: Radu Solea <radu.solea@nxp.com>
Date: Thu, 13 Jul 2017 15:00:09 +0300
Subject: [PATCH 2115/5242] MLK-14765: Enable DCP SHA workaround on all
 platforms

commit  36e97a046285f6586ea17567461f01cee8d6c008 from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove variant restriction for DCP SHA workaround. All integrations of
DCP seem affected.

Signed-off-by: Radu Solea <radu.solea@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/mxs-dcp.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/crypto/mxs-dcp.c b/drivers/crypto/mxs-dcp.c
index c713a51..92d5aab 100644
--- a/drivers/crypto/mxs-dcp.c
+++ b/drivers/crypto/mxs-dcp.c
@@ -1123,13 +1123,9 @@ static int mxs_dcp_probe(struct platform_device *pdev)
 	}
 
 	/*
-	 * Enable driver alignment with hw behavior on imx6
+	 * Enable driver alignment with hw behavior for sha generation
 	 */
-	if (of_machine_is_compatible("fsl,imx6sl") ||
-	    of_machine_is_compatible("fsl,imx6ull")) {
-		sdcp->enable_sha_workaround = 1;
-	} else
-		sdcp->enable_sha_workaround = 0;
+	sdcp->enable_sha_workaround = 1;
 
 	/* Create the SHA and AES handler threads. */
 	sdcp->thread[DCP_CHAN_HASH_SHA] = kthread_run(dcp_chan_thread_sha,
-- 
1.7.9.5

